web/static/js/components/surveys/SurveyShow.jsx

```diff
@@ -51,6 +51,30 @@ class SurveyShow extends Component {
       return <p>Loading...</p>
     }
 
+    let primaryMode = <div className='mode'>
+      <span className='type'>Primary Mode</span>
+      <div>
+        <i className='material-icons'>{survey.mode[0] == 'sms' ? 'sms' : 'phone'}</i>
+        <span className='mode-label name'>{survey.mode[0] == 'sms' ? 'SMS' : 'IVR'}</span>
+      </div>
+    </div>
+
+    let fallbackMode = null
+    if (survey.mode.length > 1) {
+      fallbackMode = <div className='mode'>
+        <span className='type'>Fallback Mode</span>
+        <div>
+          <i className='material-icons'>{survey.mode[1] == 'sms' ? 'sms' : 'phone'}</i>
```

Comment:
There's some repetition to produce the right icons and labels depending on mode. Extract those to functions that just depend on the string.